3party/opening_hours/opening_hours_tests/opening_hours_tests.cpp

```diff
@@ -0,0 +1,1334 @@
+/*
+ The MIT License (MIT)
+
+ Copyright (c) 2015 Mail.Ru Group
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in
+ all copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+*/
+
+#include "parse_opening_hours.hpp"
+#include "rules_evaluation.hpp"
+#include "rules_evaluation_private.hpp"
+
+#include <ctime>
+#include <iostream>
+#include <locale>
+#include <sstream>
+
+#define BOOST_TEST_MODULE OpeningHours
+
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wshorten-64-to-32"
+#include <boost/test/included/unit_test.hpp>
+#pragma clang diagnostic pop
+#else
+#include <boost/test/included/unit_test.hpp>
+#endif
+
+#include <boost/spirit/include/qi.hpp>
+
+namespace
+{
+template <typename T>
+std::string ToString(T const & t)
+{
+  std::stringstream sstr;
+  sstr << t;
+  return sstr.str();
+}
+
+template <typename Parser>
+bool Test(std::string const & str, Parser const & p, bool full_match = true)
+{
+  // We don't care about the result of the "what" function.
+  // We only care that all parsers have it:
+  boost::spirit::qi::what(p);
+
+  auto first = begin(str);
+  auto last = end(str);
+  return boost::spirit::qi::parse(first, last, p) && (!full_match || (first == last));
+}
+
+template <typename ParseResult>
+std::string ParseAndUnparse(std::string const & str)
+{
+  ParseResult parseResult;
+  if (!osmoh::Parse(str, parseResult))
+    return ":CAN'T PARSE:";
+
+  std::stringstream sstr;
+  sstr << parseResult;
+
+  return sstr.str();
+}
+
+bool GetTimeTuple(std::string const & strTime, std::string const & fmt, std::tm & tm)
+{
+  auto const rc = strptime(strTime.data(), fmt.data(), &tm);
+  return rc != nullptr;
+}
+
+struct GetTimeError: std::exception
+{
+  GetTimeError(std::string const & message): m_message(message) { }
```

Comment:
No need in space between curly braces. Also, please insert a blank line between ctor and what().